src/main/java/controller/handlers/RequestProcessingHandler.java

```diff
@@ -0,0 +1,81 @@
+package controller.handlers;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.HashMap;
+
+
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import models.Request;
+import models.RequestType;
+import services.ChatService;
+
+
+public class RequestProcessingHandler extends ChannelInboundHandlerAdapter{
+	private static ChatService service;
+	
+	public  RequestProcessingHandler() throws FileNotFoundException, IOException {
+		super();
+		service = new ChatService();
+	}
+	
+	@Override
+	public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+		// request object being decoded by RequestDecoderHandler that was registered in channel pipeline before this handler
+		Request request = (Request) msg;
+		// validate request
+		validateRequest(request);
+		
+		/*
+		 *  create arguments hashMap based on type of request. Here the same argument is used for all commands,
+		 *  so no need for checking for request type (switch block) and add arguments based on this check.
+		 */
+		HashMap<String,String> args =  new HashMap<String, String>();
+		args.put("searchKey", request.getSearchKey());
+		Object jsonRes = service.serve(request.getType().getCommandName(), args);
+
+		// create successful response
+		SuccessResponseModel response = new SuccessResponseModel();
```

Comment:
You are creating a response as if it is an HTTP response. Where are web sockets requests processed?!